client/tests/integration/multisig.rs

```diff
@@ -1,77 +1,209 @@
-            nonzero!(100_000_000_u64),
-        ))),
-        SetParameter::new(Parameter::Executor(SmartContractParameter::Fuel(nonzero!(
-            100_000_000_u64
-        )))),
-    ])?;
-
-    let account_id = ALICE_ID.clone();
-    let multisig_register_trigger_id = TriggerId::from_str("multisig_register")?;
-
-    let wasm = iroha_wasm_builder::Builder::new("../wasm_samples/multisig_register")
-        .show_output()
-        .build()?
-        .optimize()?
-        .into_bytes()?;
-    let wasm = WasmSmartContract::from_compiled(wasm);
-
-    let trigger = Trigger::new(
-        multisig_register_trigger_id.clone(),
-        Action::new(
-            wasm,
-            Repeats::Indefinitely,
-            account_id.clone(),
-            ExecuteTriggerEventFilter::new().for_trigger(multisig_register_trigger_id.clone()),
-        ),
-    );
+    let kingdom = "kingdom";
 
-    // Register trigger which would allow multisig account creation in wonderland domain
-    // Access to call this trigger shouldn't be restricted
-    test_client.submit_blocking(Register::trigger(trigger))?;
+    // Assume any domain registered after genesis
+    test_client.submit_all_blocking([Register::domain(Domain::new(kingdom.parse().unwrap()))])?;
 
-    // Create multisig account id and destroy it's private key
-    let multisig_account_id = gen_account_in("wonderland").0;
+    // One more block to generate a multisig accounts registry for the domain
+    test_client.submit_blocking(Log::new(Level::DEBUG, "Just ticking time".to_string()))?;
 
-    let multisig_trigger_id: TriggerId = format!(
-        "{}_{}_multisig_trigger",
-        multisig_account_id.signatory(),
-        multisig_account_id.domain()
-    )
-    .parse()?;
+    // Check that the multisig accounts registry has been generated
+    let multisig_accounts_registry_id = multisig_accounts_registry_of(&kingdom.parse().unwrap());
+    let _trigger = test_client
+        .query_single(client::trigger::by_id(
+            multisig_accounts_registry_id.clone(),
+        ))
+        .expect("multisig accounts registry should be generated after domain creation");
 
-    let signatories = core::iter::repeat_with(|| gen_account_in("wonderland"))
-        .take(5)
+    // Populate residents in the domain
+    let mut residents = core::iter::repeat_with(|| gen_account_in(kingdom))
+        .take(1 + N_SIGNATORIES)
         .collect::<BTreeMap<AccountId, KeyPair>>();
+    test_client.submit_all_blocking(
+        residents
+            .keys()
+            .cloned()
+            .map(Account::new)
+            .map(Register::account),
+    )?;
 
-    let args = MultisigRegisterArgs {
+    // Create a multisig account ID and discard the corresponding private key
+    // FIXME #5022 Should not allow arbitrary IDs. Otherwise, after #4426 pre-registration account will be hijacked as a multisig account
+    let multisig_account_id = gen_account_in(kingdom).0;
```

Comment:
I think the real solution is that a multisig account creation request derives a string that is a variant of multihash but without the authentication functionality, as suggested in #5022.

If all we want to do were to prevent hijacking, we can simply impose a signature on all or part of the payload of the request. However, this does not eliminate the risk that the requester may unfairly hold the private key

Reply:
Left to https://github.com/hyperledger/iroha/issues/5022#issuecomment-2390938541